spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/config/RabbitMessageChannelBinderConfiguration.java

```diff
@@ -45,15 +49,19 @@
 
 	@Autowired
 	private ConnectionFactory rabbitConnectionFactory;
-	
+
 	@Autowired
 	private RabbitBinderConfigurationProperties rabbitBinderConfigurationProperties;
-	
+
 	@Autowired
 	private SpringRabbitMQProperties springRabbitMQProperties;
 
 	@Bean
 	RabbitMessageChannelBinder rabbitMessageChannelBinder() {
+		if (rabbitConnectionFactory instanceof CachingConnectionFactory) {
+			((CachingConnectionFactory) rabbitConnectionFactory).
+					setChannelCacheSize(springRabbitMQProperties.getChannelCacheSize());
+		}
```

Comment:
This just seems wrong to me (reconfiguring Boot's object) - given that we are already overriding the `RabbitProperties` from boot auto-configure, we should really disable the boot auto-config altogether here and create our own factory.

For example, the connection factory `addresses` is populated from `RabbitProperties` but the binder `addresses` is populated from `SpringRabbitMQProperties`. Yes, the source of the property is the same but it's a bit smelly IMHO.

Either `SpringRabbitMQProperties` should just have those properties over and above those in `RabbitProperties` or we should do all the wiring here.

I think I prefer the latter, at least until (or unless) the `channelCachesize` property is added to boot.

I am just not keen on reconfiguring boot's object here.